src/include/rbd/librbd.hpp

```diff
@@ -38,6 +39,7 @@ namespace librbd {
     uint64_t id;
     uint64_t size;
     std::string name;
+    utime_t stamp;
```

Comment:
I think that is overkill at this point -- when we eventually create a new librbd v2 API that is not backwards compatible, we can look at updating the structs. You just need an API method that returns the timestamp (if populated) via an out param pointer -- if the snapshot doesn't have a timestamp, it should return -ENOENT.